r/tests/testthat/test-dplyr-arrange.R

```diff
@@ -0,0 +1,172 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+library(dplyr)
+
+tbl <- example_data_for_sorting
+
+test_that("arrange", {
+  expect_dplyr_equal(
+    input %>%
+      arrange(int, chr) %>%
+      collect(),
+    tbl %>%
+      slice_sample(prop = 1L)
```

Comment:
It took me a minute to realize that you're using `slice_sample` to shuffle the rows. What if you instead defined `tbl <- slice_sample(example_data_for_sorting, prop = 1L)` (with a comment that this is just shuffling rows) at the top of the file?